flink-java/src/main/java/org/apache/flink/api/java/utils/ParameterTool.java

```diff
@@ -91,22 +94,23 @@ public static ParameterTool fromArgs(String[] args) {
 
 			i += 1; // try to find the value
 
+			map.putIfAbsent(key, new ArrayList<>());
```

Comment:
I think by default we should preserve the old behaviour and add an overloaded version of this method `ParameterTool#fromArgs(String[] args, boolean multipleValues)` or something like that?

Reply:
I will try to add a new `MultipleParameterTool extends ParameterTool` instead. It is an enhanced class with multiple parameters support. And it will be marked as `PublicEvolving`. It should be compatible with `ParameterTool`, all the tests of `ParameterToolTest` should pass.